### DISCUSSION TITLE
Can we have separate bug-fix release cycles and new feature release cycles

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#13465] [Feature Requests]
I understand there's been some push for a bug-fix release recently whilst the release was delayed because you've all been feverishly working on dataflows! So, like semvar, could there been some short-term release cycles of bug-fix releases and then new feature releases? This would allow for smaller increment updates for bug fixes, then the bigger releases for new features!

**Comment by azrikahar** [comment]
Definitely something we'd like to have! Something akin to nightly release would also be nice, as suggested in https://github.com/directus/directus/discussions/11555#discussioncomment-2268752 in a similar discussion.

**Comment by rijkvanzanten** [comment]
> whilst the release was delayed because you've all been feverishly working on dataflows

For what it's worth, the primary reason it was delayed is that half our engineering team (myself included) was out sick from covid / the flu 🤧

**Reply by sxsws** [reply]
> For what it's worth, the primary reason it was delayed is that half our engineering team (myself included) was out sick from covid / the flu

... so you **were** feverish :laughing:

**Comment by rijkvanzanten** [comment]
What would be the ideal setup of setting something like this up? I'm imagining we'd start of with a more in-depth branching structure, where we rely on branches for every release, where a hotfix release branch would basically become a copy of the last release branch with a single commit cherry-picked on top 🤔 We should plan this out carefully, as it's pretty easy to accidentally release unintended changes / wrong builds (easy to get confused between branches if it isn't properly thought through and documented)

**Reply by sxsws** [reply]
Could bug fixes be merged immediately to master and new features be kept in a separate branch until a major release? So, the master branch will ultimately store the latest bug-fixed working version with each release being tagged from master and new feature branches being merged in at time of major release?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
